### ISSUE TITLE
How to run train a model for wikitablequestions?

### ISSUE THREAD

**Issue by entslscheia** [allenai/allennlp#3333]
In [wikitables_parser.jsonnet](https://github.com/allenai/allennlp/blob/master/training_config/wikitables_parser.jsonnet), there are several options related to the configuration for dataset (i.e., "tables_directory", "dpd_output_directory", "train_data_path" and "validation_data_path"). It also refers to the [dataset](https://ppasupat.github.io/WikiTableQuestions/), however, after I downloaded the files from the given link, I still don't know how to configure my dataset, because I didn't see either /wikitables_preprocessed/train.jsonl or /wikitables_preprocessed/validation.jsonl in the downloaded file. Do we need to do preprocessing by ourselves before using the wikitablequestions dataset?

**Comment by matt-gardner** [comment]
@pdasigi, do we have instructions for this anywhere?

**Comment by schmmd** [comment]
@matt-gardner I'd love to have usage information on the demo--would that be possible?

**Comment by pdasigi** [comment]
@entslscheia Sorry for the delayed response. Unfortunately there aren't detailed instructions anywhere for training the model. Here's an attempt:

0) The configuration file you pointed at is outdated. The current implementation in the repo is of the models described in [this paper](https://www.semanticscholar.org/paper/Iterative-Search-for-Weakly-Supervised-Semantic-Dasigi-Gardner/af17ccbdae4cbd1b67d6ab359615c8000f8fb66f). There are two models, one which relies on an approximate set of logical forms to train on (called MML in the paper), and another which relies on a pretrained model (called MBR in the paper). The training config files are [here](https://github.com/allenai/allennlp-semparse/blob/master/training_config/wikitables_mml_parser.jsonnet) and [here](https://github.com/allenai/allennlp-semparse/blob/master/training_config/wikitables_erm_parser.jsonnet) respectively. If you're trying to train a decent model, and not necessarily reproduce the best model in the paper, the MML configuration should be sufficient. These configuration files are in a different repository because we are breaking this one down into multiple application specific repos (see #2996 and #3302).

1) You need a set of approximate logical forms (those which evaluate to the correct answer for each example). You can obtain them using [this script](https://github.com/allenai/allennlp/blob/master/scripts/wikitables/search_for_logical_forms.py) like so:
`python scripts/wikitables/search_for_logical_forms.py /path/to/wikitables /path/to/wikitables/data/training.examples /path/to/logical_forms --use-agenda --output-separate-files --num-splits 10`
where `/path/to/wikitables` is the path to the unzipped official dataset directory. You can run `python scripts/wikitables/search_for_logical_forms.py -h` to see what the other arguments mean.

2) You can train the MML parser using [this configuration](https://github.com/allenai/allennlp-semparse/blob/master/training_config/wikitables_mml_parser.jsonnet) by pointing to the output of the previous step as the `offline_logical_forms_directory`, the unzipped official dataset directory as the `tables_directory`, and pointing to the appropriate train and dev sets. This should give you a decent model that on average (over all 5 splits) gets a test accuracy of about 42%.

3) (Optional) If you're trying to reproduce the performance of the best model in the paper linked above, you'll have to iteratively train MML and MBR models, with MBR models being trained using [this configuration](https://github.com/allenai/allennlp-semparse/blob/master/training_config/wikitables_erm_parser.jsonnet), with the `mml_model_file` option set to the location of the model archive resulting from training an MML model as in step 2.
